src/snowflake/cli/api/project/definition_manager.py

```diff
@@ -100,6 +101,16 @@ def _user_definition_file_if_available(project_path: Path) -> Optional[Path]:
             DefinitionManager.USER_DEFINITION_FILENAME, project_path
         )
 
+    @functools.cached_property
+    def _project_properties(self) -> ProjectProperties:
+        return load_project(self._project_config_paths)
+
     @functools.cached_property
     def project_definition(self) -> ProjectDefinition:
-        return load_project_definition(self._project_config_paths)
+        return self._project_properties.project_definition
+
+    @functools.cached_property
+    def template_context(self) -> dict:
+        definition = self._project_properties.raw_project_definition
```

Comment:
Does this mean we can use e.g. `<% ctx.native_app.name %>`? Are we documenting this or is it a "hidden feature"?

Reply:
I think we should be documenting it

Reply:
we can use this in the project file, and yes, we will be documenting it.